ecommerce/models.py

```diff
@@ -298,10 +313,13 @@ def another_user_already_redeemed(self, user):
             bool:
                 True if the coupon is not automatic and it has already been redeemed by someone else
         """
-        return not self.is_automatic and RedeemedCoupon.objects.filter(
-            coupon=self,
-            order__status=Order.FULFILLED,
-        ).exclude(order__user=user).exists()
+        return (
+            not Coupon.objects.filter(id=self.id).filter(self.is_automatic_qset()).exists() and
+            RedeemedCoupon.objects.filter(
+                coupon=self,
+                order__status=Order.FULFILLED,
+            ).exclude(order__user=user).exists()
```

Comment:
I don't understand where the redundant query is. `coupon=self` limits the query to only that coupon, and `order__status = FULFILLED` and `order.user != user` limit the query to orders fulfilled by other users